x-pack/plugins/threat_intelligence/public/modules/indicators/hooks/use_indicators.ts

```diff
@@ -70,132 +59,69 @@ export const useIndicators = ({
-        size,
-        from,
-        sort: sorting.map(({ id, direction }) => ({ [id]: direction })),
-      };
-
-      searchSubscription$.current = searchService
-        .search<IEsSearchRequest, IKibanaSearchResponse<RawIndicatorsResponse>>(
-          {
-            params: {
-              index: selectedPatterns,
-              body: requestBody,
-            },
-          },
-          {
-            abortSignal: abortController.current.signal,
-          }
-        )
-        .subscribe({
-          next: (response) => {
-            setIndicators(response.rawResponse.hits.hits);
-            setIndicatorCount(response.rawResponse.hits.total || 0);
-
-            if (isCompleteResponse(response)) {
-              setLoading(false);
-              searchSubscription$.current?.unsubscribe();
-              request.stats({}).ok({ json: response });
-              request.json(requestBody);
-            } else if (isErrorResponse(response)) {
-              setLoading(false);
-              request.error({ json: response });
-              searchSubscription$.current?.unsubscribe();
-            }
-          },
-          error: (requestError) => {
-            searchService.showError(requestError);
-            searchSubscription$.current?.unsubscribe();
-
-            if (requestError instanceof Error && requestError.name.includes('Abort')) {
-              inspectorAdapters.requests.reset();
-            } else {
-              request.error({ json: requestError });
-            }
-
-            setLoading(false);
-          },
-        });
-    },
-    [inspectorAdapters.requests, query, searchService, selectedPatterns, sorting]
-  );
-
-  const onChangeItemsPerPage = useCallback(
-    async (pageSize) => {
-      setPagination((currentPagination) => ({
-        ...currentPagination,
-        pageSize,
-        pageIndex: 0,
-      }));
-
-      loadData(0, pageSize);
-    },
-    [loadData]
+  // Go to first page after filters are changed
+  useEffect(() => {
+    onChangePage(0);
+  }, [filters, filterQuery, timeRange, sorting, onChangePage]);
+
+  const currentPageQuery = useMemo(
+    () => ({
+      timeRange,
+      filterQuery,
+      filters,
+      selectedPatterns,
+      sorting,
+      pagination,
+    }),
+    [filterQuery, filters, pagination, selectedPatterns, sorting, timeRange]
   );
 
-  const onChangePage = useCallback(
-    async (pageIndex) => {
-      setPagination((currentPagination) => ({ ...currentPagination, pageIndex }));
-      loadData(pageIndex * pagination.pageSize, pagination.pageSize);
-    },
-    [loadData, pagination.pageSize]
+  const { isLoading, isFetching, data, refetch } = useQuery(
+    ['indicatorsTable', currentPageQuery],
+    fetchIndicators({ searchService, inspectorAdapters }),
+    {
+      /**
+       * See https://tanstack.com/query/v4/docs/guides/paginated-queries
+       * This is essential for our ux
+       */
+      keepPreviousData: true,
+    }
```

Comment:
Wouldn't we have a problem here when the user clicks `refresh`?
Unless I'm mistaken, the `currentPageQuery` variable defined above wouldn't change, therefore the key passed to the `useQuery` would be similar, which means we wouldn't fetch the data again correct?

But in our case, if our timeRange is for example `{from: 'now-24h/h', to: 'now'}`, if the user clicks `refresh` let's say 1 hour after loading the page in the first place, we wouldn't query the latest data?

Reply:
well looking at the doc there is a default cache value set to 5 minutes. So if the above statement is correct it means we wouldn't get new data for 5 minutes?

Reply:
Looking at the devtools, I can see the new query is executed every time I hit the refresh button on the grid, so the cache seems to be ignored when you request the data to be reloaded specifically:)

Reply:
Update: seems that the logic behind our "refresh" button prevented the actual refetch, this is fixed now.